controllers/k8ssandra/reaper.go

```diff
@@ -253,3 +264,27 @@ func getSingleReaperDcName(kc *api.K8ssandraCluster) string {
 	}
 	return ""
 }
+
+func (r *K8ssandraClusterReconciler) addClusterToExternalReaper(
+	ctx context.Context,
+	kc *api.K8ssandraCluster,
+	actualDc *cassdcapi.CassandraDatacenter,
+	logger logr.Logger,
+) result.ReconcileResult {
+	manager := reaper.NewManager()
+	manager.SetK8sClient(r)
+	if username, password, err := manager.GetUiCredentials(ctx, kc.Spec.Reaper.UiUserSecretRef, kc.Namespace); err != nil {
+		logger.Error(err, "Failed to get Reaper UI user secret")
+		return result.RequeueSoon(r.DefaultDelay)
```

Comment:
And it's not just resurfacing the error, but also backoff timer for the requeue. So if the error keeps persisting, the operator will increase the delay for requeue to avoid unnecessary load.

Reply:
Pushed a commit fixing this.